// @sentry/nestjs/build/cjs/integrations/sentry-nest-bullmq-instrumentation.js
Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const instrumentation = require('@opentelemetry/instrumentation');
const core = require('@sentry/core');
const helpers = require('./helpers.js');

const supportedVersions = ['>=10.0.0'];
const COMPONENT = '@nestjs/bullmq';

/**
 * Custom instrumentation for nestjs bullmq module.
 *
 * This hooks into the `@Processor` class decorator, which is applied on queue processor classes.
 * It wraps the `process` method on the decorated class to fork the isolation scope for each job
 * invocation, create a span, and capture errors.
 */
class SentryNestBullMQInstrumentation extends instrumentation.InstrumentationBase {
   constructor(config = {}) {
    super('sentry-nestjs-bullmq', core.SDK_VERSION, config);
  }

  /**
   * Initializes the instrumentation by defining the modules to be patched.
   */
   init() {
    const moduleDef = new instrumentation.InstrumentationNodeModuleDefinition(COMPONENT, supportedVersions);

    moduleDef.files.push(this._getProcessorFileInstrumentation(supportedVersions));
    return moduleDef;
  }

  /**
   * Wraps the @Processor decorator.
   */
   _getProcessorFileInstrumentation(versions) {
    return new instrumentation.InstrumentationNodeModuleFile(
      '@nestjs/bullmq/dist/decorators/processor.decorator.js',
      versions,
      (moduleExports) => {
        if (instrumentation.isWrapped(moduleExports.Processor)) {
          this._unwrap(moduleExports, 'Processor');
        }
        this._wrap(moduleExports, 'Processor', this._createWrapProcessor());
        return moduleExports;
      },
      (moduleExports) => {
        this._unwrap(moduleExports, 'Processor');
      },
    );
  }

  /**
   * Creates a wrapper function for the @Processor class decorator.
   */
   _createWrapProcessor() {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return function wrapProcessor(original) {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      return function wrappedProcessor(...decoratorArgs) {
        // Extract queue name from decorator args
        // @Processor('queueName') or @Processor({ name: 'queueName' })
        const queueName =
          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
          typeof decoratorArgs[0] === 'string' ? decoratorArgs[0] : decoratorArgs[0]?.name || 'unknown';

        // Get the original class decorator
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        const classDecorator = original(...decoratorArgs);

        // Return a new class decorator that wraps the process method
        return function (target) {
          const originalProcess = target.prototype.process;

          if (
            originalProcess &&
            typeof originalProcess === 'function' &&
            !target.__SENTRY_INTERNAL__ &&
            !originalProcess.__SENTRY_INSTRUMENTED__
          ) {
            target.prototype.process = new Proxy(originalProcess, {
              apply: (originalProcessFn, thisArg, args) => {
                return core.withIsolationScope(() => {
                  return core.startSpan(helpers.getBullMQProcessSpanOptions(queueName), async () => {
                    try {
                      return await originalProcessFn.apply(thisArg, args);
                    } catch (error) {
                      core.captureException(error, {
                        mechanism: {
                          handled: false,
                          type: 'auto.queue.nestjs.bullmq',
                        },
                      });
                      throw error;
                    }
                  });
                });
              },
            });

            target.prototype.process.__SENTRY_INSTRUMENTED__ = true;
          }

          // Apply the original class decorator
          // eslint-disable-next-line @typescript-eslint/no-unsafe-return
          return classDecorator(target);
        };
      };
    };
  }
}

exports.SentryNestBullMQInstrumentation = SentryNestBullMQInstrumentation;
