// tencentcloud-sdk-nodejs@4.1.216/es/services/hai/v20230812/hai_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("hai.tencentcloudapi.com", "2023-08-12", clientConfig);
    }
    async DescribeServiceLoginSettings(req, cb) {
        return this.request("DescribeServiceLoginSettings", req, cb);
    }
    async ResizeInstanceDisk(req, cb) {
        return this.request("ResizeInstanceDisk", req, cb);
    }
    async DescribeDeployTemplates(req, cb) {
        return this.request("DescribeDeployTemplates", req, cb);
    }
    async TerminateInstances(req, cb) {
        return this.request("TerminateInstances", req, cb);
    }
    async DeleteService(req, cb) {
        return this.request("DeleteService", req, cb);
    }
    async UpdateServiceConfigs(req, cb) {
        return this.request("UpdateServiceConfigs", req, cb);
    }
    async InquirePriceRunInstances(req, cb) {
        return this.request("InquirePriceRunInstances", req, cb);
    }
    async InquirePriceUpdateServiceConfigs(req, cb) {
        return this.request("InquirePriceUpdateServiceConfigs", req, cb);
    }
    async DescribeInstances(req, cb) {
        return this.request("DescribeInstances", req, cb);
    }
    async RunInstances(req, cb) {
        return this.request("RunInstances", req, cb);
    }
    async DeployInferService(req, cb) {
        return this.request("DeployInferService", req, cb);
    }
    async DescribeServicesCallInfo(req, cb) {
        return this.request("DescribeServicesCallInfo", req, cb);
    }
    async StartInstance(req, cb) {
        return this.request("StartInstance", req, cb);
    }
    async ResetInstancesPassword(req, cb) {
        return this.request("ResetInstancesPassword", req, cb);
    }
    async DescribeMuskPrompts(req, cb) {
        return this.request("DescribeMuskPrompts", req, cb);
    }
    async DescribeRegions(req, cb) {
        return this.request("DescribeRegions", req, cb);
    }
    async DescribeServices(req, cb) {
        return this.request("DescribeServices", req, cb);
    }
    async CreateMuskPrompt(req, cb) {
        return this.request("CreateMuskPrompt", req, cb);
    }
    async StopInstance(req, cb) {
        return this.request("StopInstance", req, cb);
    }
    async CreateApplication(req, cb) {
        return this.request("CreateApplication", req, cb);
    }
    async DescribeModels(req, cb) {
        return this.request("DescribeModels", req, cb);
    }
    async DescribeApplications(req, cb) {
        return this.request("DescribeApplications", req, cb);
    }
    async DescribeScenes(req, cb) {
        return this.request("DescribeScenes", req, cb);
    }
    async DescribeInstanceNetworkStatus(req, cb) {
        return this.request("DescribeInstanceNetworkStatus", req, cb);
    }
    async CreateInferServiceByTemplate(req, cb) {
        return this.request("CreateInferServiceByTemplate", req, cb);
    }
}
