// tencentcloud-sdk-nodejs@4.1.216/es/services/config/v20220802/config_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("config.tencentcloudapi.com", "2022-08-02", clientConfig);
    }
    async AddConfigRule(req, cb) {
        return this.request("AddConfigRule", req, cb);
    }
    async DescribeCompliancePack(req, cb) {
        return this.request("DescribeCompliancePack", req, cb);
    }
    async StartAggregateConfigRuleEvaluation(req, cb) {
        return this.request("StartAggregateConfigRuleEvaluation", req, cb);
    }
    async UpdateAggregateConfigDeliver(req, cb) {
        return this.request("UpdateAggregateConfigDeliver", req, cb);
    }
    async ListDiscoveredResources(req, cb) {
        return this.request("ListDiscoveredResources", req, cb);
    }
    async ListAggregators(req, cb) {
        return this.request("ListAggregators", req, cb);
    }
    async ListConfigRules(req, cb) {
        return this.request("ListConfigRules", req, cb);
    }
    async OpenConfigRule(req, cb) {
        return this.request("OpenConfigRule", req, cb);
    }
    async CreateAggregator(req, cb) {
        return this.request("CreateAggregator", req, cb);
    }
    async OpenConfigRecorder(req, cb) {
        return this.request("OpenConfigRecorder", req, cb);
    }
    async StartConfigRuleEvaluation(req, cb) {
        return this.request("StartConfigRuleEvaluation", req, cb);
    }
    async DeleteAlarmPolicy(req, cb) {
        return this.request("DeleteAlarmPolicy", req, cb);
    }
    async AddAggregateConfigRule(req, cb) {
        return this.request("AddAggregateConfigRule", req, cb);
    }
    async ListAggregateDiscoveredResources(req, cb) {
        return this.request("ListAggregateDiscoveredResources", req, cb);
    }
    async CreateRemediation(req, cb) {
        return this.request("CreateRemediation", req, cb);
    }
    async ListSystemRules(req, cb) {
        return this.request("ListSystemRules", req, cb);
    }
    async DeleteConfigRule(req, cb) {
        return this.request("DeleteConfigRule", req, cb);
    }
    async ListResourceTypes(req, cb) {
        return this.request("ListResourceTypes", req, cb);
    }
    async UpdateRemediation(req, cb) {
        return this.request("UpdateRemediation", req, cb);
    }
    async UpdateCompliancePack(req, cb) {
        return this.request("UpdateCompliancePack", req, cb);
    }
    async StartRemediation(req, cb) {
        return this.request("StartRemediation", req, cb);
    }
    async DescribeConfigDeliver(req, cb) {
        return this.request("DescribeConfigDeliver", req, cb);
    }
    async ListRemediations(req, cb) {
        return this.request("ListRemediations", req, cb);
    }
    async UpdateConfigDeliver(req, cb) {
        return this.request("UpdateConfigDeliver", req, cb);
    }
    async DeleteAggregateCompliancePack(req, cb) {
        return this.request("DeleteAggregateCompliancePack", req, cb);
    }
    async ListAggregateCompliancePacks(req, cb) {
        return this.request("ListAggregateCompliancePacks", req, cb);
    }
    async DescribeAggregateConfigRule(req, cb) {
        return this.request("DescribeAggregateConfigRule", req, cb);
    }
    async DescribeConfigRecorder(req, cb) {
        return this.request("DescribeConfigRecorder", req, cb);
    }
    async ListAlarmPolicy(req, cb) {
        return this.request("ListAlarmPolicy", req, cb);
    }
    async UpdateAggregateConfigRule(req, cb) {
        return this.request("UpdateAggregateConfigRule", req, cb);
    }
    async UpdateCompliancePackStatus(req, cb) {
        return this.request("UpdateCompliancePackStatus", req, cb);
    }
    async AddAlarmPolicy(req, cb) {
        return this.request("AddAlarmPolicy", req, cb);
    }
    async DescribeAggregateCompliancePack(req, cb) {
        return this.request("DescribeAggregateCompliancePack", req, cb);
    }
    async OpenAggregateConfigRule(req, cb) {
        return this.request("OpenAggregateConfigRule", req, cb);
    }
    async DescribeAggregateConfigDeliver(req, cb) {
        return this.request("DescribeAggregateConfigDeliver", req, cb);
    }
    async DetachConfigRuleToCompliancePack(req, cb) {
        return this.request("DetachConfigRuleToCompliancePack", req, cb);
    }
    async DescribeConfigRule(req, cb) {
        return this.request("DescribeConfigRule", req, cb);
    }
    async ListAggregateConfigRuleEvaluationResults(req, cb) {
        return this.request("ListAggregateConfigRuleEvaluationResults", req, cb);
    }
    async AddCompliancePack(req, cb) {
        return this.request("AddCompliancePack", req, cb);
    }
    async CloseConfigRecorder(req, cb) {
        return this.request("CloseConfigRecorder", req, cb);
    }
    async DescribeDiscoveredResource(req, cb) {
        return this.request("DescribeDiscoveredResource", req, cb);
    }
    async UpdateAlarmPolicy(req, cb) {
        return this.request("UpdateAlarmPolicy", req, cb);
    }
    async DescribeAggregator(req, cb) {
        return this.request("DescribeAggregator", req, cb);
    }
    async UpdateConfigRecorder(req, cb) {
        return this.request("UpdateConfigRecorder", req, cb);
    }
    async DeleteRemediations(req, cb) {
        return this.request("DeleteRemediations", req, cb);
    }
    async PutEvaluations(req, cb) {
        return this.request("PutEvaluations", req, cb);
    }
    async UpdateAggregateCompliancePack(req, cb) {
        return this.request("UpdateAggregateCompliancePack", req, cb);
    }
    async DescribeSystemCompliancePack(req, cb) {
        return this.request("DescribeSystemCompliancePack", req, cb);
    }
    async UpdateConfigRule(req, cb) {
        return this.request("UpdateConfigRule", req, cb);
    }
    async DescribeSystemRule(req, cb) {
        return this.request("DescribeSystemRule", req, cb);
    }
    async CloseAggregateConfigRule(req, cb) {
        return this.request("CloseAggregateConfigRule", req, cb);
    }
    async DetachAggregateConfigRuleToCompliancePack(req, cb) {
        return this.request("DetachAggregateConfigRuleToCompliancePack", req, cb);
    }
    async ListRemediationExecutions(req, cb) {
        return this.request("ListRemediationExecutions", req, cb);
    }
    async ListAggregateConfigRules(req, cb) {
        return this.request("ListAggregateConfigRules", req, cb);
    }
    async ListSystemCompliancePacks(req, cb) {
        return this.request("ListSystemCompliancePacks", req, cb);
    }
    async ListConfigRuleEvaluationResults(req, cb) {
        return this.request("ListConfigRuleEvaluationResults", req, cb);
    }
    async AddAggregateCompliancePack(req, cb) {
        return this.request("AddAggregateCompliancePack", req, cb);
    }
    async DescribeAggregateDiscoveredResource(req, cb) {
        return this.request("DescribeAggregateDiscoveredResource", req, cb);
    }
    async UpdateAggregateCompliancePackStatus(req, cb) {
        return this.request("UpdateAggregateCompliancePackStatus", req, cb);
    }
    async CloseConfigRule(req, cb) {
        return this.request("CloseConfigRule", req, cb);
    }
    async ListCompliancePacks(req, cb) {
        return this.request("ListCompliancePacks", req, cb);
    }
    async DeleteCompliancePack(req, cb) {
        return this.request("DeleteCompliancePack", req, cb);
    }
    async DeleteAggregateConfigRule(req, cb) {
        return this.request("DeleteAggregateConfigRule", req, cb);
    }
}
