// tencentcloud-sdk-nodejs/tencentcloud/services/config/v20220802/config_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars */
/*
 * Copyright (c) 2018 Tencent. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
const abstract_client_1 = require("../../../common/abstract_client");
/**
 * config client
 * @class
 */
class Client extends abstract_client_1.AbstractClient {
    constructor(clientConfig) {
        super("config.tencentcloudapi.com", "2022-08-02", clientConfig);
    }
    /**
     * 新建 规则
     */
    async AddConfigRule(req, cb) {
        return this.request("AddConfigRule", req, cb);
    }
    /**
     * 合规包详情
     */
    async DescribeCompliancePack(req, cb) {
        return this.request("DescribeCompliancePack", req, cb);
    }
    /**
     * 账号组触发评估
     */
    async StartAggregateConfigRuleEvaluation(req, cb) {
        return this.request("StartAggregateConfigRuleEvaluation", req, cb);
    }
    /**
     * 账号组编辑投递设置
     */
    async UpdateAggregateConfigDeliver(req, cb) {
        return this.request("UpdateAggregateConfigDeliver", req, cb);
    }
    /**
     * 获取资源列表
     */
    async ListDiscoveredResources(req, cb) {
        return this.request("ListDiscoveredResources", req, cb);
    }
    /**
     * 账号组列表
     */
    async ListAggregators(req, cb) {
        return this.request("ListAggregators", req, cb);
    }
    /**
     * 获取规则列表
     */
    async ListConfigRules(req, cb) {
        return this.request("ListConfigRules", req, cb);
    }
    /**
     * 开启规则
     */
    async OpenConfigRule(req, cb) {
        return this.request("OpenConfigRule", req, cb);
    }
    /**
     * 新建账号组
     */
    async CreateAggregator(req, cb) {
        return this.request("CreateAggregator", req, cb);
    }
    /**
     * 资源监控管理-开启监控
     */
    async OpenConfigRecorder(req, cb) {
        return this.request("OpenConfigRecorder", req, cb);
    }
    /**
     * 触发评估
     */
    async StartConfigRuleEvaluation(req, cb) {
        return this.request("StartConfigRuleEvaluation", req, cb);
    }
    /**
     * 删除告警规则
     */
    async DeleteAlarmPolicy(req, cb) {
        return this.request("DeleteAlarmPolicy", req, cb);
    }
    /**
     * 账号组新建规则
     */
    async AddAggregateConfigRule(req, cb) {
        return this.request("AddAggregateConfigRule", req, cb);
    }
    /**
     * 账号组获取资源列表
     */
    async ListAggregateDiscoveredResources(req, cb) {
        return this.request("ListAggregateDiscoveredResources", req, cb);
    }
    /**
     * 新增规则修正设置
     */
    async CreateRemediation(req, cb) {
        return this.request("CreateRemediation", req, cb);
    }
    /**
     * 获取预设规则列表
     */
    async ListSystemRules(req, cb) {
        return this.request("ListSystemRules", req, cb);
    }
    /**
     * 删除规则
     */
    async DeleteConfigRule(req, cb) {
        return this.request("DeleteConfigRule", req, cb);
    }
    /**
     * 获取资源类型列表
     */
    async ListResourceTypes(req, cb) {
        return this.request("ListResourceTypes", req, cb);
    }
    /**
     * 新增告警监控规则
     */
    async UpdateRemediation(req, cb) {
        return this.request("UpdateRemediation", req, cb);
    }
    /**
     * 编辑合规包
     */
    async UpdateCompliancePack(req, cb) {
        return this.request("UpdateCompliancePack", req, cb);
    }
    /**
     * 手动执行规则修复
     */
    async StartRemediation(req, cb) {
        return this.request("StartRemediation", req, cb);
    }
    /**
     * 获取投递设置详情
     */
    async DescribeConfigDeliver(req, cb) {
        return this.request("DescribeConfigDeliver", req, cb);
    }
    /**
     * 修正设置列表
     */
    async ListRemediations(req, cb) {
        return this.request("ListRemediations", req, cb);
    }
    /**
     * 编辑投递设置
     */
    async UpdateConfigDeliver(req, cb) {
        return this.request("UpdateConfigDeliver", req, cb);
    }
    /**
     * 账号组删除合规包
     */
    async DeleteAggregateCompliancePack(req, cb) {
        return this.request("DeleteAggregateCompliancePack", req, cb);
    }
    /**
     * 账号组获取合规包列表
     */
    async ListAggregateCompliancePacks(req, cb) {
        return this.request("ListAggregateCompliancePacks", req, cb);
    }
    /**
     * 账号组获取规则详情
     */
    async DescribeAggregateConfigRule(req, cb) {
        return this.request("DescribeAggregateConfigRule", req, cb);
    }
    /**
     * 获取监控详情
     */
    async DescribeConfigRecorder(req, cb) {
        return this.request("DescribeConfigRecorder", req, cb);
    }
    /**
     * 告警规则列表
     */
    async ListAlarmPolicy(req, cb) {
        return this.request("ListAlarmPolicy", req, cb);
    }
    /**
     * 账号组编辑规则
     */
    async UpdateAggregateConfigRule(req, cb) {
        return this.request("UpdateAggregateConfigRule", req, cb);
    }
    /**
     * 开启、关闭合规包
     */
    async UpdateCompliancePackStatus(req, cb) {
        return this.request("UpdateCompliancePackStatus", req, cb);
    }
    /**
     * 新增告警监控规则
     */
    async AddAlarmPolicy(req, cb) {
        return this.request("AddAlarmPolicy", req, cb);
    }
    /**
     * 账号组合规包详情
     */
    async DescribeAggregateCompliancePack(req, cb) {
        return this.request("DescribeAggregateCompliancePack", req, cb);
    }
    /**
     * 账号组开启规则
     */
    async OpenAggregateConfigRule(req, cb) {
        return this.request("OpenAggregateConfigRule", req, cb);
    }
    /**
     * 账号组获取投递设置详情
     */
    async DescribeAggregateConfigDeliver(req, cb) {
        return this.request("DescribeAggregateConfigDeliver", req, cb);
    }
    /**
     * 合规包移除规则
     */
    async DetachConfigRuleToCompliancePack(req, cb) {
        return this.request("DetachConfigRuleToCompliancePack", req, cb);
    }
    /**
     * 获取规则详情
     */
    async DescribeConfigRule(req, cb) {
        return this.request("DescribeConfigRule", req, cb);
    }
    /**
     * 账号组获取评估结果--规则维度（某个规则下资源的评估结果列表）
     */
    async ListAggregateConfigRuleEvaluationResults(req, cb) {
        return this.request("ListAggregateConfigRuleEvaluationResults", req, cb);
    }
    /**
     * 新建合规包
     */
    async AddCompliancePack(req, cb) {
        return this.request("AddCompliancePack", req, cb);
    }
    /**
     * 资源监控管理-关闭监控
     */
    async CloseConfigRecorder(req, cb) {
        return this.request("CloseConfigRecorder", req, cb);
    }
    /**
     * 资源详情
     */
    async DescribeDiscoveredResource(req, cb) {
        return this.request("DescribeDiscoveredResource", req, cb);
    }
    /**
     * 更新告警规则
     */
    async UpdateAlarmPolicy(req, cb) {
        return this.request("UpdateAlarmPolicy", req, cb);
    }
    /**
     * 账号组详情
     */
    async DescribeAggregator(req, cb) {
        return this.request("DescribeAggregator", req, cb);
    }
    /**
     * 编辑监控范围
     */
    async UpdateConfigRecorder(req, cb) {
        return this.request("UpdateConfigRecorder", req, cb);
    }
    /**
     * 删除规则修正设置
     */
    async DeleteRemediations(req, cb) {
        return this.request("DeleteRemediations", req, cb);
    }
    /**
     * 上报自定义规则评估结果
     */
    async PutEvaluations(req, cb) {
        return this.request("PutEvaluations", req, cb);
    }
    /**
     * 账号组编辑合规包
     */
    async UpdateAggregateCompliancePack(req, cb) {
        return this.request("UpdateAggregateCompliancePack", req, cb);
    }
    /**
     * 获取系统合规包详情
     */
    async DescribeSystemCompliancePack(req, cb) {
        return this.request("DescribeSystemCompliancePack", req, cb);
    }
    /**
     * 编辑规则
     */
    async UpdateConfigRule(req, cb) {
        return this.request("UpdateConfigRule", req, cb);
    }
    /**
     * 预设规则详情
     */
    async DescribeSystemRule(req, cb) {
        return this.request("DescribeSystemRule", req, cb);
    }
    /**
     * 账号组关闭规则
     */
    async CloseAggregateConfigRule(req, cb) {
        return this.request("CloseAggregateConfigRule", req, cb);
    }
    /**
     * 账号组合规包移除规则
     */
    async DetachAggregateConfigRuleToCompliancePack(req, cb) {
        return this.request("DetachAggregateConfigRuleToCompliancePack", req, cb);
    }
    /**
     * 修正记录
     */
    async ListRemediationExecutions(req, cb) {
        return this.request("ListRemediationExecutions", req, cb);
    }
    /**
     * 账号组获取规则列表
     */
    async ListAggregateConfigRules(req, cb) {
        return this.request("ListAggregateConfigRules", req, cb);
    }
    /**
     * 获取系统合规包列表
     */
    async ListSystemCompliancePacks(req, cb) {
        return this.request("ListSystemCompliancePacks", req, cb);
    }
    /**
     *  获取评估结果--规则维度（某个规则下资源的评估结果列表）
     */
    async ListConfigRuleEvaluationResults(req, cb) {
        return this.request("ListConfigRuleEvaluationResults", req, cb);
    }
    /**
     * 账号组新建合规包
     */
    async AddAggregateCompliancePack(req, cb) {
        return this.request("AddAggregateCompliancePack", req, cb);
    }
    /**
     * 账号组资源详情
     */
    async DescribeAggregateDiscoveredResource(req, cb) {
        return this.request("DescribeAggregateDiscoveredResource", req, cb);
    }
    /**
     * 账号组开启、关闭合规包
     */
    async UpdateAggregateCompliancePackStatus(req, cb) {
        return this.request("UpdateAggregateCompliancePackStatus", req, cb);
    }
    /**
     * 关闭规则
     */
    async CloseConfigRule(req, cb) {
        return this.request("CloseConfigRule", req, cb);
    }
    /**
     * 获取合规包列表
     */
    async ListCompliancePacks(req, cb) {
        return this.request("ListCompliancePacks", req, cb);
    }
    /**
     * 删除合规包
     */
    async DeleteCompliancePack(req, cb) {
        return this.request("DeleteCompliancePack", req, cb);
    }
    /**
     * 账号组删除规则
     */
    async DeleteAggregateConfigRule(req, cb) {
        return this.request("DeleteAggregateConfigRule", req, cb);
    }
}
exports.Client = Client;
