// tencentcloud-sdk-nodejs@4.1.216/tencentcloud/services/apis/v20240801/apis_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars */
/*
 * Copyright (c) 2018 Tencent. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
const abstract_client_1 = require("../../../common/abstract_client");
/**
 * apis client
 * @class
 */
class Client extends abstract_client_1.AbstractClient {
    constructor(clientConfig) {
        super("apis.tencentcloudapi.com", "2024-08-01", clientConfig);
    }
    /**
     * 查询Credential详情
     */
    async DescribeAgentCredential(req, cb) {
        return this.request("DescribeAgentCredential", req, cb);
    }
    /**
     * 查询app列表
     */
    async DescribeAgentApps(req, cb) {
        return this.request("DescribeAgentApps", req, cb);
    }
    /**
     * 查询模型详情
     */
    async DescribeModel(req, cb) {
        return this.request("DescribeModel", req, cb);
    }
    /**
     * 删除app的mcp server
     */
    async DeleteAgentAppMcpServers(req, cb) {
        return this.request("DeleteAgentAppMcpServers", req, cb);
    }
    /**
     * 查询模型服务详情
     */
    async DescribeModelService(req, cb) {
        return this.request("DescribeModelService", req, cb);
    }
    /**
     * 查询Credential列表
     */
    async DescribeAgentCredentials(req, cb) {
        return this.request("DescribeAgentCredentials", req, cb);
    }
    /**
     * 创建app的mcp server关联
     */
    async CreateAgentAppMcpServers(req, cb) {
        return this.request("CreateAgentAppMcpServers", req, cb);
    }
    /**
     * 查询app详情
     */
    async DescribeAgentApp(req, cb) {
        return this.request("DescribeAgentApp", req, cb);
    }
    /**
     * 查询app mcpServer关联列表
     */
    async DescribeAgentAppMcpServers(req, cb) {
        return this.request("DescribeAgentAppMcpServers", req, cb);
    }
    /**
     * 修改模型
     */
    async ModifyModel(req, cb) {
        return this.request("ModifyModel", req, cb);
    }
    /**
     * 查询mcp server详情
     */
    async DescribeMcpServer(req, cb) {
        return this.request("DescribeMcpServer", req, cb);
    }
    /**
     * 修改Credential
     */
    async ModifyAgentCredential(req, cb) {
        return this.request("ModifyAgentCredential", req, cb);
    }
    /**
     * 删除mcp server
     */
    async DeleteMcpServer(req, cb) {
        return this.request("DeleteMcpServer", req, cb);
    }
    /**
     * 查询mcp server列表
     */
    async DescribeMcpServers(req, cb) {
        return this.request("DescribeMcpServers", req, cb);
    }
    /**
     * 修改mcp server
     */
    async ModifyMcpServer(req, cb) {
        return this.request("ModifyMcpServer", req, cb);
    }
    /**
     * 创建app
     */
    async CreateAgentApp(req, cb) {
        return this.request("CreateAgentApp", req, cb);
    }
    /**
     * 查询app modelService关联列表
     */
    async DescribeAgentAppModelServices(req, cb) {
        return this.request("DescribeAgentAppModelServices", req, cb);
    }
    /**
     * 删除app的model service关联
     */
    async DeleteAgentAppModelServices(req, cb) {
        return this.request("DeleteAgentAppModelServices", req, cb);
    }
    /**
     * 创建模型服务
     */
    async CreateModelService(req, cb) {
        return this.request("CreateModelService", req, cb);
    }
    /**
     * 创建Credential
     */
    async CreateAgentCredential(req, cb) {
        return this.request("CreateAgentCredential", req, cb);
    }
    /**
     * 删除Credential
     */
    async DeleteAgentCredential(req, cb) {
        return this.request("DeleteAgentCredential", req, cb);
    }
    /**
     * 创建app的model service关联
     */
    async CreateAgentAppModelServices(req, cb) {
        return this.request("CreateAgentAppModelServices", req, cb);
    }
    /**
     * 删除模型服务
     */
    async DeleteModelService(req, cb) {
        return this.request("DeleteModelService", req, cb);
    }
    /**
     * 查询模型列表
     */
    async DescribeModels(req, cb) {
        return this.request("DescribeModels", req, cb);
    }
    /**
     * 修改app
     */
    async ModifyAgentApp(req, cb) {
        return this.request("ModifyAgentApp", req, cb);
    }
    /**
     * 修改模型服务
     */
    async ModifyModelService(req, cb) {
        return this.request("ModifyModelService", req, cb);
    }
    /**
     * 删除app
     */
    async DeleteAgentApp(req, cb) {
        return this.request("DeleteAgentApp", req, cb);
    }
    /**
     * 创建模型
     */
    async CreateModel(req, cb) {
        return this.request("CreateModel", req, cb);
    }
    /**
     * 查询模型服务列表
     */
    async DescribeModelServices(req, cb) {
        return this.request("DescribeModelServices", req, cb);
    }
    /**
     * 编辑app的model service关联
     */
    async ModifyAgentAppModelServices(req, cb) {
        return this.request("ModifyAgentAppModelServices", req, cb);
    }
    /**
     * 创建mcp server
     */
    async CreateMcpServer(req, cb) {
        return this.request("CreateMcpServer", req, cb);
    }
    /**
     * 删除模型
     */
    async DeleteModel(req, cb) {
        return this.request("DeleteModel", req, cb);
    }
}
exports.Client = Client;
