// @sentry/nestjs@10.49.0/build/cjs/integrations/sentry-nest-event-instrumentation.js
Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const instrumentation = require('@opentelemetry/instrumentation');
const core = require('@sentry/core');
const helpers = require('./helpers.js');

const supportedVersions = ['>=2.0.0'];
const COMPONENT = '@nestjs/event-emitter';

/**
 * Custom instrumentation for nestjs event-emitter
 *
 * This hooks into the `OnEvent` decorator, which is applied on event handlers.
 * Wrapped handlers run inside a forked isolation scope to ensure event-scoped data
 * (breadcrumbs, tags, etc.) does not leak between concurrent event invocations
 * or into subsequent HTTP requests.
 */
class SentryNestEventInstrumentation extends instrumentation.InstrumentationBase {
   constructor(config = {}) {
    super('sentry-nestjs-event', core.SDK_VERSION, config);
  }

  /**
   * Initializes the instrumentation by defining the modules to be patched.
   */
   init() {
    const moduleDef = new instrumentation.InstrumentationNodeModuleDefinition(COMPONENT, supportedVersions);

    moduleDef.files.push(this._getOnEventFileInstrumentation(supportedVersions));
    return moduleDef;
  }

  /**
   * Wraps the @OnEvent decorator.
   */
   _getOnEventFileInstrumentation(versions) {
    return new instrumentation.InstrumentationNodeModuleFile(
      '@nestjs/event-emitter/dist/decorators/on-event.decorator.js',
      versions,
      (moduleExports) => {
        if (instrumentation.isWrapped(moduleExports.OnEvent)) {
          this._unwrap(moduleExports, 'OnEvent');
        }
        this._wrap(moduleExports, 'OnEvent', this._createWrapOnEvent());
        return moduleExports;
      },
      (moduleExports) => {
        this._unwrap(moduleExports, 'OnEvent');
      },
    );
  }

  /**
   * Creates a wrapper function for the @OnEvent decorator.
   */
   _createWrapOnEvent() {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return function wrapOnEvent(original) {
      return function wrappedOnEvent(event, options) {
        // Get the original decorator result
        const decoratorResult = original(event, options);

        // Return a new decorator function that wraps the handler
        return (target, propertyKey, descriptor) => {
          if (
            !descriptor.value ||
            typeof descriptor.value !== 'function' ||
            target.__SENTRY_INTERNAL__ ||
            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
            descriptor.value.__SENTRY_INSTRUMENTED__
          ) {
            return decoratorResult(target, propertyKey, descriptor);
          }

          function eventNameFromEvent(event) {
            if (typeof event === 'string') {
              return event;
            } else if (Array.isArray(event)) {
              return event.map(eventNameFromEvent).join(',');
            } else return String(event);
          }

          const originalHandler = descriptor.value;
          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
          const handlerName = originalHandler.name || propertyKey;
          let eventName = eventNameFromEvent(event);

          // Instrument the actual handler
          descriptor.value = async function (...args) {
            // When multiple @OnEvent decorators are used on a single method, we need to get all event names
            // from the reflector metadata as there is no information during execution which event triggered it
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            // @ts-ignore - reflect-metadata of nestjs adds these methods to Reflect
            if (Reflect.getMetadataKeys(descriptor.value).includes('EVENT_LISTENER_METADATA')) {
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore - reflect-metadata of nestjs adds these methods to Reflect
              const eventData = Reflect.getMetadata('EVENT_LISTENER_METADATA', descriptor.value);
              if (Array.isArray(eventData)) {
                eventName = eventData
                  .map((data) => {
                    if (data && typeof data === 'object' && 'event' in data && data.event) {
                      return eventNameFromEvent(data.event);
                    }
                    return '';
                  })
                  .reverse() // decorators are evaluated bottom to top
                  .join('|');
              }
            }

            return core.withIsolationScope(() => {
              return core.startSpan(helpers.getEventSpanOptions(eventName), async () => {
                try {
                  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
                  const result = await originalHandler.apply(this, args);
                  return result;
                } catch (error) {
                  // exceptions from event handlers are not caught by global error filter
                  core.captureException(error, {
                    mechanism: {
                      handled: false,
                      type: 'auto.event.nestjs',
                    },
                  });
                  throw error;
                }
              });
            });
          };

          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
          descriptor.value.__SENTRY_INSTRUMENTED__ = true;

          // Preserve the original function name
          Object.defineProperty(descriptor.value, 'name', {
            value: handlerName,
            configurable: true,
          });

          // Apply the original decorator
          return decoratorResult(target, propertyKey, descriptor);
        };
      };
    };
  }
}

exports.SentryNestEventInstrumentation = SentryNestEventInstrumentation;
