// tencentcloud-sdk-nodejs@4.1.216/es/services/apis/v20240801/apis_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("apis.tencentcloudapi.com", "2024-08-01", clientConfig);
    }
    async DescribeAgentCredential(req, cb) {
        return this.request("DescribeAgentCredential", req, cb);
    }
    async DescribeAgentApps(req, cb) {
        return this.request("DescribeAgentApps", req, cb);
    }
    async DescribeModel(req, cb) {
        return this.request("DescribeModel", req, cb);
    }
    async DeleteAgentAppMcpServers(req, cb) {
        return this.request("DeleteAgentAppMcpServers", req, cb);
    }
    async DescribeModelService(req, cb) {
        return this.request("DescribeModelService", req, cb);
    }
    async DescribeAgentCredentials(req, cb) {
        return this.request("DescribeAgentCredentials", req, cb);
    }
    async CreateAgentAppMcpServers(req, cb) {
        return this.request("CreateAgentAppMcpServers", req, cb);
    }
    async DescribeAgentApp(req, cb) {
        return this.request("DescribeAgentApp", req, cb);
    }
    async DescribeAgentAppMcpServers(req, cb) {
        return this.request("DescribeAgentAppMcpServers", req, cb);
    }
    async ModifyModel(req, cb) {
        return this.request("ModifyModel", req, cb);
    }
    async DescribeMcpServer(req, cb) {
        return this.request("DescribeMcpServer", req, cb);
    }
    async ModifyAgentCredential(req, cb) {
        return this.request("ModifyAgentCredential", req, cb);
    }
    async DeleteMcpServer(req, cb) {
        return this.request("DeleteMcpServer", req, cb);
    }
    async DescribeMcpServers(req, cb) {
        return this.request("DescribeMcpServers", req, cb);
    }
    async ModifyMcpServer(req, cb) {
        return this.request("ModifyMcpServer", req, cb);
    }
    async CreateAgentApp(req, cb) {
        return this.request("CreateAgentApp", req, cb);
    }
    async DescribeAgentAppModelServices(req, cb) {
        return this.request("DescribeAgentAppModelServices", req, cb);
    }
    async DeleteAgentAppModelServices(req, cb) {
        return this.request("DeleteAgentAppModelServices", req, cb);
    }
    async CreateModelService(req, cb) {
        return this.request("CreateModelService", req, cb);
    }
    async CreateAgentCredential(req, cb) {
        return this.request("CreateAgentCredential", req, cb);
    }
    async DeleteAgentCredential(req, cb) {
        return this.request("DeleteAgentCredential", req, cb);
    }
    async CreateAgentAppModelServices(req, cb) {
        return this.request("CreateAgentAppModelServices", req, cb);
    }
    async DeleteModelService(req, cb) {
        return this.request("DeleteModelService", req, cb);
    }
    async DescribeModels(req, cb) {
        return this.request("DescribeModels", req, cb);
    }
    async ModifyAgentApp(req, cb) {
        return this.request("ModifyAgentApp", req, cb);
    }
    async ModifyModelService(req, cb) {
        return this.request("ModifyModelService", req, cb);
    }
    async DeleteAgentApp(req, cb) {
        return this.request("DeleteAgentApp", req, cb);
    }
    async CreateModel(req, cb) {
        return this.request("CreateModel", req, cb);
    }
    async DescribeModelServices(req, cb) {
        return this.request("DescribeModelServices", req, cb);
    }
    async ModifyAgentAppModelServices(req, cb) {
        return this.request("ModifyAgentAppModelServices", req, cb);
    }
    async CreateMcpServer(req, cb) {
        return this.request("CreateMcpServer", req, cb);
    }
    async DeleteModel(req, cb) {
        return this.request("DeleteModel", req, cb);
    }
}
